index.js

```diff
@@ -187,16 +198,21 @@ function session(options) {
     // the store has temporarily disconnected etc
     if (!storeReady) {
       debug('store is disconnected')
+      cleanEventListeners();
```

Comment:
So if a request comes in when the store is not ready, and this removes the listener to make it ready again, then how will the session ever work again? It seems like if the store disconnects and if one request just happened to come in while disconnected, express-session will then never think the store is ever ready again until the entire process is restarted? That seems like a large issue. But I am just starting to take a look at the changes, so maybe I am mistaken. I marked tests are needed, as we'll want to have tests that cover all these new code paths added and probably the scenario I described above too.